### PR TITLE
Remove most methods from GTMSessionFetcherServiceProtocol.

### PR DESCRIPTION
**Pull request by mwyman**
Cleans up the protocol to exclude extraneous methods, and makes GTMSessionFetcher know about GTMSessionFetcherService explicitly; this separation was originally to permit only building `GTMSessionFetcher.{h,m}`, but this is pretty much never done anymore. It's now necessary to build the fetcher and the service.

Creates internal headers for `GTMSessionFetcher` and `GTMSessionFetcherService` that contain some methods that don't need to be exposed externally.

This makes it possible to update interaction between the fetcher and the service without demanding the protocol be updated and possibly change required methods for projects that are faking/mocking the service to overwrite fetching behavior.

### CODE REVIEW

**Comment by thomasvl** [comment]
Removing myself from review request since you are still working on this.  Let me know when it's ready for a review.

File: Sources/Core/Public/GTMSessionFetcher/GTMSessionFetcher.h [hunk 1]
```diff
@@ -953,7 +932,7 @@ __deprecated_msg("implement GTMSessionFetcherAuthorizer instead")
 #pragma clang diagnostic pop
 
 // The service object that created and monitors this fetcher, if any.
-@property(atomic, strong) id<GTMSessionFetcherServiceProtocol> service;
+@property(atomic, strong) GTMSessionFetcherService *service;
```
[the remark this answers is not in the document]
Comment (mwyman) [reply Sources/Core/Public/GTMSessionFetcher/GTMSessionFetcher.h:935]: Unfortunately I've seen folks create fetchers with resume data, and provide the service afterwards. I contemplated adding a `fetcherWithResumeData:` on the service, but it's not clear they always want all the other callbacks/etc to be added on.

File: Package.swift [hunk 2]
```diff
@@ -49,7 +49,10 @@ let package = Package(
         .testTarget(
             name: "GTMSessionFetcherCoreTests",
             dependencies: ["GTMSessionFetcherFull", "GTMSessionFetcherCore"],
-            path: "UnitTests"
+            path: "UnitTests",
+            cSettings: [
```
[the remark this answers is not in the document]
Comment (thomasvl) [reply Package.swift:53]: Sorry, was thinking this was more than the test target, guess that's not too bad.

**Review by thomasvl — APPROVED** [review]
Maybe keep the methods SignIn use until that PR lands. But good to go otherwise.

**Review by thomasvl — APPROVED** [review]

**Pull request merged**